homeassistant/components/medcom_ble/config_flow.py

```diff
@@ -0,0 +1,170 @@
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfo,
+    async_discovered_service_info,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class Discovery:
+    """A discovered bluetooth device."""
+
+    name: str
+    discovery_info: BluetoothServiceInfo
+    device: MedcomBleDevice
+
+
+def get_name(device: MedcomBleDevice) -> str:
+    """Generate name with model and identifier for device."""
+
+    name = device.friendly_name()
+    if identifier := device.identifier:
+        name += f" ({identifier})"
+    return name
+
+
+class MedcomBleDeviceUpdateError(Exception):
+    """Custom error class for device updates."""
+
+
+class InspectorBLEConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Medcom BLE radiation monitors."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovered_device: Discovery | None = None
+        self._discovered_devices: dict[str, Discovery] = {}
+
+    async def _get_device_data(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> MedcomBleDevice:
+        ble_device = bluetooth.async_ble_device_from_address(
+            self.hass, discovery_info.address
+        )
+        if ble_device is None:
+            _LOGGER.debug("no ble_device in _get_device_data")
+            raise MedcomBleDeviceUpdateError("No ble_device")
+
+        inspector = MedcomBleDeviceData(_LOGGER)
+
+        try:
+            data = await inspector.update_device(ble_device)
+        except BleakError as err:
+            _LOGGER.error(
+                "Error connecting to and getting data from %s: %s",
+                discovery_info.address,
+                err,
+            )
+            raise MedcomBleDeviceUpdateError("Failed getting device data") from err
+        except Exception as err:
+            _LOGGER.error(
+                "Unknown error occurred from %s: %s", discovery_info.address, err
+            )
+            raise err
+        return data
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug("Discovered BLE device: %s", discovery_info)
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+
+        try:
+            device = await self._get_device_data(discovery_info)
+        except MedcomBleDeviceUpdateError:
+            return self.async_abort(reason="cannot_connect")
+        except Exception:  # pylint: disable=broad-except
+            return self.async_abort(reason="unknown")
+
```

Comment:
With bluetooth, we shouldn't try to connect to the device until they confirm they want to set it up since Bluetooth has a limited number of connection slots